### PR TITLE
First steps for interpreter

### PR DESCRIPTION
**Pull request by qwe-q**
Implements a couple instructions directly in memory. Can't open files yet

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,3 +1,157 @@
+struct Mips {
+    regs: [u32; 32],
+    floats: [f32; 32],
+    mult_hi: u32,
+    mult_lo: u32,
+    pc: usize,
+
+    // Enough for six MIPS instructions.
+    memory: [u32; 13]
```
Comment (CharlesAverill) [inline src/main.rs:9]: Memory size will be constant across architectures, so 13 here should be some global constant

File: src/main.rs [hunk 2]
```diff
@@ -1,3 +1,157 @@
...
+}
+
+struct Rtype {
+    rs: usize,
+    rt: usize,
+    rd: usize,
+    shamt: u8,
+    funct: u8
+}
```
Comment (CharlesAverill) [inline src/main.rs:18]: Note for later: we should just merge our impls.
Also nitpick but we shouldn't use a `usize` for something that isn't denoting a size

File: src/main.rs [hunk 3]
```diff
@@ -1,3 +1,157 @@
...
+            // Subtract
+            0x22 => {
+                //Todo- catch overflows
+                self.regs[ins.rd] = self.regs[ins.rt] + self.regs[ins.rs];
+            }
+            // Xor
+            0x26 => {
+                self.regs[ins.rd] = self.regs[ins.rt] ^ self.regs[ins.rs];
+            }
+            _ => panic!("R-Type unimplemented instruction")
```
Comment (CharlesAverill) [inline src/main.rs:62]: Didn't we say no panics?

**Review by CharlesAverill — CHANGES_REQUESTED** [review]
Tiny things

File: src/main.rs (hunk 1 above)
Comment (qwe-q) [reply src/main.rs:9]: yeah, this should be changed to a Vec for all memory regions.

File: src/main.rs (hunk 2 above)
Comment (qwe-q) [reply src/main.rs:18]: I'm still unconvinced that writing ``u5`` and ``u6`` types was a good idea. I'd prefer not to implement them into my codebase. Also, keeping the assembler and disassembler decoupled feels worth having a more in-depth discussion over.

re: usize: I can change that.

File: src/main.rs (hunk 3 above)
Comment (qwe-q) [reply src/main.rs:62]: we did. I need to plumb errors in everywhere.

**Comment by qwe-q** [comment]
I'm currently making some pretty substantial changes to this code, so I'm going to close this PR and re-open when it's ready for merging again.

**Pull request closed without merging**